lib/mayaUsd/utils/diagnosticDelegate.cpp

```diff
@@ -68,7 +74,8 @@ static MString _FormatDiagnostic(const TfDiagnosticBase& d)
         d.GetContext().GetFunction(),
         d.GetContext().GetLine(),
         d.GetContext().GetFile());
-    return msg.c_str();
+
+    return TfGetEnvSetting(MAYAUSD_SHOW_FULL_DIAGNOSTICS) ? msg.c_str() : d.GetCommentary().c_str();
```

Comment:
Most of the time we will not be in full diagnostics mode. With the current code structure, we will always generate a full string. String operations are costly, how about we insert if/else condition and only generate msg string when needed?

Reply:
Yes agreed. I also wonder about the need to check this new env var on every call for _FormatDiagnostic(). Do we need to be able to modify this env var during a Maya run? How about just getting this var once into a static variable?

Reply:
@kxl-adsk @seando-adsk I agree with you that std::string allocation can be costly and am very well aware of them. But the cost here is super minimal. I honestly don't believe we don't need to worry about operation costs in here. We can discuss this internally if needed.

Reply:
I would be happy to see the profiler results to understand where we hit this issue.